File: Semester 4/Programming and Projection Enironments/mpp-proiect-java/Proiect_MPP/AirportPersistence/src/main/java/root/persistence/airports/AirportRepository.java
package root.persistence.airports;

import root.model.Airport;
import root.persistence.AbstractRepository;

import java.sql.*;
import java.util.Collection;
import java.util.Properties;

public class AirportRepository extends AbstractRepository<Airport, Integer> implements IAirportRepository {

    public AirportRepository(Properties properties) {
        super(properties);
        logger.info("Initializing Airport Repository ");
    }

    @Override
    protected Airport extractEntity(ResultSet resultSet) throws SQLException {
        int id = resultSet.getInt("id");
        String name = resultSet.getString("name");
        String cityName = resultSet.getString("city_name");
        return new Airport(id, name, cityName);
    }

    @Override
    protected PreparedStatement createInsertStatement(Connection connection, Airport entity) throws SQLException {
        PreparedStatement statement = connection.prepareStatement(super.sqlCommand, Statement.RETURN_GENERATED_KEYS);
        statement.setInt(1, entity.getId());
        statement.setString(2, entity.getName());
        statement.setString(3, entity.getCityName());
        return statement;
    }

    @Override
    protected PreparedStatement createUpdateStatement(Connection connection, Airport entity, Integer integer) throws SQLException {
        PreparedStatement statement = connection.prepareStatement(super.sqlCommand);
        statement.setInt(3, entity.getId());
        statement.setString(1, entity.getName());
        statement.setString(2, entity.getCityName());
        return statement;
    }

    @Override
    protected PreparedStatement createDeleteStatement(Connection connection, Airport entity) throws SQLException {
        PreparedStatement statement = connection.prepareStatement(super.sqlCommand);
        statement.setLong(1, entity.getId());
        return statement;
    }

    @Override
    public int add(Airport elem) {
        super.sqlCommand = "INSERT INTO airports(id, name, city_name) VALUES (?, ?, ?)";
        return super.add(elem);
    }

    @Override
    public void delete(Airport elem) {
        super.sqlCommand = "DELETE FROM airports WHERE id=?";
        super.delete(elem);
    }

    @Override
    public void update(Airport elem, Integer integer) {
        super.sqlCommand = "UPDATE airports SET name=?, city_name=? WHERE id=?";
        super.update(elem, integer);
    }

    @Override
    public Airport findById(Integer integer) {
        super.sqlCommand = "SELECT * FROM airports WHERE id="+integer;
        return super.getOne();
    }

    @Override
    public Collection<Airport> getAll() {
        super.sqlCommand = "SELECT * FROM airports";
        return super.getAll();
    }

    @Override
    public Collection<Airport> getAirportAfterName(String name) {
        super.sqlCommand = "SELECT * FROM airports WHERE name='" + name + '\'';
        return super.getAll();
    }

    @Override
    public Collection<Airport> getAirportsInCity(String cityName) {
        super.sqlCommand = "SELECT * FROM airports WHERE LOWER(city_name) LIKE '" + cityName.toLowerCase() + "%'";
        logger.info(super.sqlCommand);
        return super.getAll();
    }

}

File: Semester 4/Programming and Projection Enironments/mpp-proiect-csharp/Proiect_MPP/persistence/airports/AirportRepository.cs
using System.Data.SQLite;
using model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Data;
using persistence.databaseUtils;

namespace persistence.airports
{
    public class AirportRepository : AbstractRepository<Airport, int>, IAirportRepository
    {
        public AirportRepository(IDictionary<string, string> properties) : base(properties)
        {
            logger.Info("Initializing Airport Repository.");
        }
        protected override Airport extractEntity(IDataReader dataReader)
        {
            int id = dataReader.GetInt32(0);
            string name = dataReader.GetString(1);
            string cityName = dataReader.GetString(2);
            return new Airport(id, name, cityName);
        }

        public override int add(Airport item)
        {
            base.connection = DbUtils.getConnection(base.properties);
            using (base.sqlCommand = connection.CreateCommand())
            {
                base.sqlCommand.CommandText = "INSERT INTO airports(name, city_name) VALUES (@name, @cityName)";

                var paramName = base.sqlCommand.CreateParameter();
                paramName.ParameterName = "@name";
                paramName.Value = item.Name;
                base.sqlCommand.Parameters.Add(paramName);

                var paramCityName = base.sqlCommand.CreateParameter();
                paramCityName.ParameterName = "@id";
                paramCityName.Value = item.CityName;
                base.sqlCommand.Parameters.Add(paramCityName);

                return base.add(item);
            }
        }

        public override void delete(Airport item)
        {
            base.connection = DbUtils.getConnection(base.properties);
            using (base.sqlCommand = connection.CreateCommand())
            {
                base.sqlCommand.CommandText = "DELETE FROM airports WHERE id=@id";

                var paramId = base.sqlCommand.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = item.ID;
                base.sqlCommand.Parameters.Add(paramId);

                base.delete(item);
            }
        }
        public override void update(Airport item, int id)
        {
            base.connection = DbUtils.getConnection(base.properties);
            using (base.sqlCommand = connection.CreateCommand())
            {
                base.sqlCommand.CommandText = "UPDATE airports SET name=@name, city_name=@cityName WHERE id=@id";

                var paramId = base.sqlCommand.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = item.ID;
                base.sqlCommand.Parameters.Add(paramId);

                var paramName = base.sqlCommand.CreateParameter();
                paramName.ParameterName = "@name";
                paramName.Value = item.Name;
                base.sqlCommand.Parameters.Add(paramName);

                var paramCityName = base.sqlCommand.CreateParameter();
                paramCityName.ParameterName = "@id";
                paramCityName.Value = item.CityName;
                base.sqlCommand.Parameters.Add(paramCityName);

                base.update(item, id);
            }
        }

        public override List<Airport> findAll()
        {
            base.connection = DbUtils.getConnection(base.properties);
            using (base.sqlCommand = connection.CreateCommand())
            {
                base.sqlCommand.CommandText = "SELECT * FROM airports";

                return base.findAll();
            }
        }

        public override Airport? findById(int id)
        {
            base.connection = DbUtils.getConnection(base.properties);
            using (base.sqlCommand = connection.CreateCommand())
            {
                base.sqlCommand.CommandText = "SELECT * FROM airports WHERE id=@id";
                var paramId = base.sqlCommand.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                base.sqlCommand.Parameters.Add(paramId);

                return base.findById(id);
            }
        }

        public List<Airport> getAirportAfterName(string name)
        {
            base.connection = DbUtils.getConnection(base.properties);
            using (base.sqlCommand = connection.CreateCommand())
            {
                base.sqlCommand.CommandText = "SELECT * FROM airports WHERE name=@name";

                var paramName = base.sqlCommand.CreateParameter();
                paramName.ParameterName = "@name";
                paramName.Value = name;
                base.sqlCommand.Parameters.Add(paramName);

                return base.findAll();
            }
        }

        public List<Airport> getAirportsInCity(string cityName)
        {
            base.connection = DbUtils.getConnection(base.properties);
            using (base.sqlCommand = connection.CreateCommand())
            {
                base.sqlCommand.CommandText = "SELECT * FROM airports WHERE LOWER(city_name) LIKE @cityName";
                logger.Info(base.sqlCommand.CommandText);

                var paramName = base.sqlCommand.CreateParameter();
                paramName.ParameterName = "@cityName";
                paramName.Value = cityName.ToLower() + '%';
                base.sqlCommand.Parameters.Add(paramName);

                return base.findAll();
            }
        }
    }
}
